orchagent/fabricportsorch.cpp

```diff
@@ -147,32 +152,96 @@ bool FabricPortsOrch::allPortsReady()
 
 void FabricPortsOrch::generatePortStats()
 {
-    // FIX_ME: This function installs flex counters for port stats
-    // on fabric ports for fabric asics and voq asics (that connect
-    // to fabric asics via fabric ports). These counters will be
-    // installed in FLEX_COUNTER_DB, and queried by syncd and updated
-    // to COUNTERS_DB.
-    // However, currently BCM SAI doesn't update its code to query
-    // port stats (metrics in list port_stat_ids) yet.
-    // Also, BCM sets too low value for "Max logical port count" (256),
-    // causing syncd to crash on voq asics that now include regular front
-    // panel ports, fabric ports, and multiple logical ports.
-    // So, this function will just do nothing for now, and we will readd
-    // code to install port stats counters when BCM completely supports.
+    if (!m_fabricPortStatEnabled) return;
+
+    SWSS_LOG_NOTICE("Generate fabric port stats");
+
+    vector<FieldValueTuple> portNamePortCounterMap;
+    for (auto p : m_fabricLanePortMap)
+    {
+        int lane = p.first;
+        sai_object_id_t port = p.second;
+
+        std::ostringstream portName;
+        portName << FABRIC_PORT_PREFIX << lane;
+        portNamePortCounterMap.emplace_back(portName.str(), sai_serialize_object_id(port));
+
+        // Install flex counters for port stats
+        std::unordered_set<std::string> counter_stats;
+        for (const auto& it: port_stat_ids)
+        {
+            counter_stats.emplace(sai_serialize_port_stat(it));
+        }
+        port_stat_manager.setCounterIdList(port, CounterType::PORT, counter_stats);
+    }
+    m_portNamePortCounterTable->set("", portNamePortCounterMap);
 }
 
 void FabricPortsOrch::generateQueueStats()
```

Comment:
Is function `generateQueueStats` used anywhere ?

Reply:
`generateQueueStats` is called by `FlexCounterOrch`. This code was already in place.